Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp

```diff
@@ -558,144 +533,66 @@ int CLuaColShapeDefs::GetColPolygonPoints(lua_State* luaVM)
         return 1;
     }
 
-    argStream.SetCustomError("ColShape must be Polygon");
+    argStream.SetCustomError("ColShape must be polygon");
     return luaL_error(luaVM, argStream.GetFullErrorMessage());
 }
 
-int CLuaColShapeDefs::GetColPolygonPointPosition(lua_State* luaVM)
+const CVector2D& CLuaColShapeDefs::GetColPolygonPointPosition_OOP(CClientColShape* const colShape, unsigned int pointIndex)
 {
-    CClientColShape* pColShape;
-    uint             uiPointIndex;
-
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pColShape);
-    argStream.ReadNumber(uiPointIndex);
-
-    if (argStream.HasErrors())
-        return luaL_error(luaVM, argStream.GetFullErrorMessage());
-
-    if (pColShape->GetShapeType() == COLSHAPE_POLYGON)
+    if (colShape->GetShapeType() == COLSHAPE_POLYGON)
     {
-        CClientColPolygon* pColPolygon = static_cast<CClientColPolygon*>(pColShape);
-        CVector2D          vecPoint;
-        if (uiPointIndex > 0 && CStaticFunctionDefinitions::GetColPolygonPointPosition(pColPolygon, uiPointIndex - 1, vecPoint))
-        {
-            lua_pushvector(luaVM, vecPoint);
-        }
-        else
+        pointIndex--; // Lua indexing starts at 1, we start at 0.
+
+        CClientColPolygon* const colPolygon = static_cast<CClientColPolygon*>(colShape);
+        if (pointIndex < colPolygon->CountPoints())
         {
-            m_pScriptDebugging->LogWarning(luaVM, "Invalid point index");
-            lua_pushboolean(luaVM, false);
+            return *(colPolygon->IterBegin() + pointIndex);
         }
-        return 1;
     }
-
-    argStream.SetCustomError("ColShape must be Polygon");
-    return luaL_error(luaVM, argStream.GetFullErrorMessage());
 }
 
-int CLuaColShapeDefs::SetColPolygonPointPosition(lua_State* luaVM)
+bool CLuaColShapeDefs::SetColPolygonPointPosition(CClientColShape* const colShape, unsigned int pointIndex, const CVector2D point)
 {
-    CClientColShape* pColShape;
-    uint             uiPointIndex;
-    CVector2D        vecPoint;
+    if (colShape->GetShapeType() != COLSHAPE_POLYGON)
+        throw std::invalid_argument("ColShape must be a polygon");
 
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pColShape);
-    argStream.ReadNumber(uiPointIndex);
-    argStream.ReadVector2D(vecPoint);
+        pointIndex--; // lua starts at 1, we at 0
```

Comment:
```suggestion
    pointIndex--; // lua starts at 1, we at 0
```

Should there be an extra indent here?